File: Alpro III/T1 Alpro III/T1.cpp
#include <iostream>
#include <ctime>

using namespace std;

int main(void)
{
	for (int monkey = 1; monkey < 7; ++monkey)
	{
		clock_t start;
		double ttime;
		start = clock();
		int crew = 9;
		bool ready;
		long unsigned mulberrys, morningMulberry, individualPortion;
		for (int i = 1; ; ++i)
		{
			mulberrys = i * crew;
			morningMulberry = mulberrys;
			ready = true;
			for (int j = 0; j < crew; ++j)
			{
				if (mulberrys % (crew - 1) != 0)
				{
					ready = false;
					break;
				}
				individualPortion = mulberrys / (crew - 1);
				mulberrys += individualPortion + monkey;
			}
			if (ready)
			{
				break;
			}
		}
		ttime = (clock() - start) / (double) CLOCKS_PER_SEC;
		cout << "Mulberrys in the morning: " << morningMulberry << endl;
		cout << "Mulberrys: " << mulberrys << endl;
		cout << "Time: " << ttime << "s" << endl;
		for (int i = 1; i <= crew; ++i)
		{
			cout << "T" << i << ": " << mulberrys/crew + morningMulberry/crew<< endl;
			mulberrys = mulberrys - mulberrys/crew - monkey;
		}
		cout << endl;
	}
	return 0;
}

File: Alpro III/T1 Alpro III/T1.java
public class T1
{
	public static final int CREW = 11;
	//public static final int MONKEY_FINAL = 6;

	public static void main(String[] args)
	{
		for (long monkey = 1; monkey < 7; monkey++)
		{
			long mulberrys = 0;
			//long monkey = MONKEY_FINAL;
			long remaining = 0;
			long morningMullberry;
			long begin = System.currentTimeMillis();
			for (remaining = 1; ; remaining++)
			{
				mulberrys = remaining * CREW;
				morningMullberry = mulberrys;
				boolean ok = true;
				for (int t = 0; t < CREW; t++)
				{
					if (mulberrys % (CREW - 1) != 0)
					{
						ok = false;
						break;
					}
					long individualPart = mulberrys / (CREW - 1);
					mulberrys += individualPart + monkey;
				}
				if (ok)
				{
					break;
				}
			}
			long end = System.currentTimeMillis();
			System.out.println("Crew: " + CREW);
			System.out.println("Monkey: " + monkey);
			System.out.println("Mulberrys " + mulberrys);
			System.out.println("MorningMullberry: " + morningMullberry);
			System.out.println(end - begin + "ms");
			for (int i = 1; i <= CREW; i++)
			{
				System.out.println("T" + i + ": " + (mulberrys/CREW + morningMullberry/CREW));
				mulberrys = mulberrys - mulberrys/CREW - monkey;
			}
			System.out.println();
		}
	}
}